packages/expo-splash-screen/ios/SplashScreenManager.swift

```diff
@@ -0,0 +1,92 @@
+import React
+import UIKit
+import ExpoModulesCore
+
+public class SplashScreenManager: NSObject {
+  @objc public static let shared = SplashScreenManager()
+  private var loadingView: UIView?
+  private var rootView: UIView?
+  private var options = SplashScreenOptions()
+  public var preventAutoHideCalled = false
+
+  private override init() {}
+
+  public func initWith(_ rootView: UIView) {
+    if RCTRunningInAppExtension() {
+      return
+    }
+
+    self.rootView = rootView
+    if let vc = UIStoryboard(name: "SplashScreen", bundle: nil).instantiateInitialViewController() {
+      loadingView = vc.view
+      loadingView?.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+
+      if let bounds = self.rootView?.bounds {
+        loadingView?.frame = bounds
+        loadingView?.center = CGPoint(x: bounds.midX, y: bounds.midY)
+      }
+      loadingView?.isHidden = false
+#if RCT_NEW_ARCH_ENABLED
+      if let hostView = rootView as? RCTSurfaceHostingProxyRootView, let loadingView {
+        hostView.disableActivityIndicatorAutoHide(true)
+        hostView.loadingView = loadingView
+      }
+#else
+      if let loadingView {
+        self.rootView?.addSubview(loadingView)
+      }
+#endif
+    }
+
+    NotificationCenter.default.addObserver(self, selector: #selector(onAppReady), name: Notification.Name("RCTContentDidAppearNotification"), object: nil)
+    NotificationCenter.default.addObserver(self, selector: #selector(onAppReady), name: Notification.Name.RCTJavaScriptDidLoad, object: nil)
```

Comment:
I think it's more logic to only fire `onAppReady` / auto `hide` only on `RCTContentDidAppearNotification`